Fw/Sm/SmSignalBuffer.hpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * SmSignalBuffer.hpp
+ *
+ */
+
+/*
+ * Description:
+ * This object contains the SmSignalBuffer type, used for attaching data to state machine signals
+ */
+#ifndef FW_SM_SIGNAL_BUFFER_HPP
+#define FW_SM_SIGNAL_BUFFER_HPP
+
+#include <FpConfig.hpp>
+#include <Fw/Types/Serializable.hpp>
+
+namespace Fw {
+
+    class SmSignalBuffer : public SerializeBufferBase {
+        public:
+
+            enum {
+                SERIALIZED_TYPE_ID = 1010,
+                SERIALIZED_SIZE = FW_COM_BUFFER_MAX_SIZE + sizeof(FwBuffSizeType)  // size of buffer + storage of size word
```

Comment:
In new code, instead of FwBuffSizeType we should use FwSizeStoreType.

Reply:
Done